include/tvm/relay/expr.h

```diff
@@ -152,6 +152,9 @@ class Var;
 /*! \brief Container for Var */
 class VarNode : public ExprNode {
  public:
+  VarNode(Id vid, Type type_annotation) : vid(vid), type_annotation(type_annotation) {}
+  VarNode() : vid("") {}
```

Comment:
The not-null that involves composite objects are more complicated and this is not necessarily the best approach, let us wait a bit before we find a better solution

Reply:
> The not-null that involves composite objects are more complicated and this is not necessarily the best approach, let us wait a bit before we find a better solution

@tqchen : Thanks! I also felt that this is not best way. We might have to change Node registration framework i believe. Will wait for your future guidance!